biz.aQute.repository/src/aQute/maven/provider/MavenRepository.java

```diff
@@ -58,7 +59,10 @@ public MavenRepository(File base, String id, List<MavenBackingRepository> releas
 
 		this.executor = new PromiseExecutor(executor);
 		this.localOnly = this.release.isEmpty() && this.snapshot.isEmpty();
-		IO.mkdirs(base);
+		if (Files.isSymbolicLink(this.base.toPath()))
```

Comment:
I would suggest that instead we change `IO.mkdirs` to:

```java
	public static void mkdirs(Path dir) throws IOException {
		if (Files.isSymbolicLink(dir)) {
			mkdirs(Files.readSymbolicLink(dir));
		} else {
			Files.createDirectories(dir);
		}
	}
```
 Which correctly handles symlinks which point to symlinks which point to a real folder.

Putting the fix in IO means everyone benefits. The tests should move to the aQute.lib project too.

Reply:
Not sure about the best way to address your comment. I'm going to push second commit on top of the original one. Is it OK? Or is it better to amend an existing commit?

Reply:
Just amend your commit and push force as we discussed at EclipseCon.